app/controllers/admin_api/books_controller.rb

```diff
@@ -53,7 +59,19 @@ def book_params
     params.require(:book).permit(:title, :author, :published_date, :price, :category)
   end
 
+  def discount_params
+    params.permit(:discount_amount, :discount_type)
+  end
+
   def set_book
     @book = Book.find(params[:id])
   end
+
+  def process_discounts
+    if params[:discount_amount] == ""
+      @book.remove_discount
```

Comment:
This would probably be more clear if written:
```
unless params[:discount_amount]
  @book.apply...
else
  @book.remove...
end
```

If you don't agree, no biggie.